components/dialog/stories/template.js

```diff
@@ -1,95 +1,190 @@
+// import { getRandomId, renderContent } from "@spectrum-css/preview/decorators";
 import { html } from "lit";
 import { classMap } from "lit/directives/class-map.js";
 import { ifDefined } from "lit/directives/if-defined.js";
+import { styleMap } from "lit/directives/style-map.js";
 import { when } from "lit/directives/when.js";
 
-import { Template as Button } from "@spectrum-css/button/stories/template.js";
-import { Template as CloseButton } from "@spectrum-css/closebutton/stories/template.js";
-import { Template as Divider } from "@spectrum-css/divider/stories/template.js";
-import { Template as Modal } from "@spectrum-css/modal/stories/template.js";
-import { Template as Underlay } from "@spectrum-css/underlay/stories/template.js";
-
 import "../index.css";
 
 export const Template = ({
 	rootClass = "spectrum-Dialog",
-	isDismissable = true,
+	isDismissible = false,
 	isOpen = true,
 	showModal = false,
+	hasFooter = false,
 	heading,
+	header,
+	footer,
+	hasCheckbox = false,
 	content = [],
 	customClasses = [],
 	id,
-	...globals
-}) => {
-	const { scale } = globals;
-	const [, updateArgs] = useArgs();
+	size = "m",
+	layout,
+	heroImageUrl,
+	customStyles = {},
+} = {}, context = {}) => {
+	const { updateArgs } = context;
+	const toggleOpen = () => updateArgs({ isOpen: !isOpen });
 
 	const Dialog = html`
 		<div
 			class=${classMap({
 				[rootClass]: true,
-				[`${rootClass}--${scale}`]: true,
-				[`${rootClass}--dismissable`]: isDismissable,
+				[`${rootClass}--dismissible`]: isDismissible && ["fullscreen", "fullscreenTakeover"].every(l => layout !== l),
+				[`${rootClass}--${layout}`]: typeof layout !== "undefined",
+				[`${rootClass}--size${size?.toUpperCase()}`]: typeof size !== "undefined", 
 				...customClasses.reduce((a, c) => ({ ...a, [c]: true }), {}),
 			})}
 			id=${ifDefined(id)}
 			role="dialog"
 			tabindex="-1"
 			aria-modal="true"
-		>
+			style=${ifDefined(styleMap(customStyles))}
+		> 
 			<div class="${rootClass}-grid">
-				${when(heading, () => [
-					html`<h1 class="${rootClass}-heading">${heading}</h1>`,
-					Divider({
-						horizontal: true,
-						customClasses: [`${rootClass}-divider`],
-						...globals,
-					}),
-				])}
+				${when(typeof heroImageUrl !== "undefined", () =>
+					html`
+						<div 
+							class="spectrum-Dialog-hero"
+							style="background-image:url(${heroImageUrl})">
+						</div>
+					`
+				)}
+				<div class="${rootClass}-header">
+					${when(heading, () => html`
+						<h1 class="${rootClass}-heading">${heading}</h1>
+					`)}
+					${when(header, () => html`
+						<span class="${rootClass}-header-content">
+							${Typography({
+								semantics: "body",
+								size: "m",
+								// @todo: takeover dialogs can accept other components in their headers. could the renderContent function work here?
+								content: [ header ]
+							})}
+						</span>
+					`,
+					() => html`
+						<span class="${rootClass}-heading--noHeader"></span>
```

Comment:
I meant to combine these two commits, but I believe I've gotten rid of the `--noHeader` styles that weren't really doing anything. 

https://github.com/adobe/spectrum-css/pull/2860/commits/fd8c087703e31b722872e1fa9ce3b3131218dbd6
https://github.com/adobe/spectrum-css/pull/2860/commits/836e3a754099ff2fd5e589727b9b27771cf1ffc1

Reply:
Thanks for getting rid of that class and element that are no longer needed.

I'm seeing something going on with the grid when isDismissible is true. It looks like the header needs to expand over other column? Should the grid template area be `". b c c c g g"` instead of `. b c c . g g`?:
<img width="646" alt="Screenshot 2024-11-15 at 2 20 20 PM" src="https://github.com/user-attachments/assets/cd96c5c3-8f27-410f-b618-401298a12a2a">
<img width="668" alt="Screenshot 2024-11-15 at 2 18 55 PM" src="https://github.com/user-attachments/assets/c7c4a623-e738-4f7a-bd18-c57de850b3da">